scripts/build.sh

```diff
@@ -5,63 +5,109 @@
 ALL_XC_ARCH="386 amd64 arm arm64 ppc64le"
 ALL_XC_OS="linux darwin windows freebsd openbsd solaris"
 
+# Exit immediately if a command fails
 set -e
 
+# Validates that a necessary tool is on the PATH
+function validateToolPresence
+{
+    local TOOLNAME=$1
+    which ${TOOLNAME} >/dev/null || echo "${TOOLNAME} is not on the path. Exiting..."
+    exit 1
+}
+
+# Validates that all used tools are present; exits when any is not found
+function validatePreconditions
+{
+    echo "==> Checking for necessary tools..."
+    validateToolPresence realpath
+    validateToolPresence dirname
+    validateToolPresence tr
+    validateToolPresence find
+}
+
 # Get the parent directory of where this script is.
-SOURCE="${BASH_SOURCE[0]}"
-while [ -h "$SOURCE" ] ; do SOURCE="$(readlink "$SOURCE")"; done
-DIR="$( cd -P "$( dirname "$SOURCE" )/.." && pwd )"
-
-# Change into that directory
-cd $DIR
-
-# Delete the old dir
-echo "==> Removing old directory..."
-rm -f bin/*
-rm -rf pkg/*
-mkdir -p bin/
-
-# helpers for Cygwin-hosted builds
-: ${OSTYPE:=`uname`}
-
-case $OSTYPE in
-    MINGW*|MSYS*|cygwin|CYGWIN*)
-        # cygwin only translates ';' to ':' on select environment variables
-        PATHSEP=';'
-        ;;
-    *)	PATHSEP=':'
-esac
-
-function convert_path() {
-    local flag
-    [ "${1:0:1}" = '-' ] && { flag="$1"; shift; }
+# NOTE: I'm unsure why you don't just use realpath like below
+function enterPackerSourceDir
```

Comment:
Functions replacing the inline code.
The "enterPackerSourceDir" function now uses realpath instead of readlink. More readable?